db.select: Adding JSON and CSV support

gulshan-123 opened the pull request:
This is a new PR instead of polluting the commits in #6721.

Fixes #6390  

Used parson wrapper of grass to format the output as json.
Currently, only plain and json format is supported. will add csv and vertical option too after the first review as major portion was to be done only for json support, rest are straightforward using parameter settings.

CC: @petrasovaa

---
Demo output
```bash
grass > db.select driver=sqlite database=/tmp/testdb.sqlite sql="SELECT agency, AVG(score) FROM firestations GROUP BY agency" format=json
{
    "info": {
        "columns": [
            {
                "name": "agency",
                "sql_type": "TEXT",
                "is_number": false
            },
            {
                "name": "AVG(score)",
                "sql_type": "DOUBLE PRECISION",
                "is_number": true
            }
        ]
    },
    "records": [
        {
            "agency": "FD",
            "AVG(score)": 9.5833333333333304
        }
    ]
}
```

---
TODO: 
- [x] Add JSON support
- [x] Add CSV, and vertical format in OutputFormat
- [x] Add CSV as default instead of Plain as suggested in original issue, @petrasovaa, please confirm if the default separator should be comma `,` (general CSV) or pipe `|` (currently default).
- [x] Add tests

petrasovaa commented on db/db.select/main.c:
```diff
@@ -135,7 +208,8 @@ int sel(dbDriver *driver, dbString *stmt)
     }
 
-    if (parms.output && strcmp(parms.output, "-") != 0) {
+    if (parms.format != JSON && parms.output &&
```
The JSON should be output to the requested file.

gulshan-123 commented:
Thank you for the review, I have added all the suggested change in my latest commit, also I have tested format options.

- Added support for format=csv and format=vertical too, along with earlier format=json and format=plain
- By default, format=plain will be used with separator as `|`.
- I have also made some options mutually exclusive
- Added a verbose message when -v flag will be used, similar to v.category handling of -g flag.

Please review once again, and if everything is fine, I will move ahead for adding tests... 

Also since, db.select has no folder named testsuite, as in others like v.db.select, should I create one? or tests for db.select will be written somewhere else?
I believe tests for db.select to be of similar to v.db.select

petrasovaa commented on db/db.select/main.c:
```diff
@@ -302,6 +469,23 @@ void parse_command_line(int argc, char **argv)
+    if (v->answer) {
+        G_verbose_message(
+            _("Flag 'v' is deprecated and will be removed in a future "
+              "release. Please use format=vertical instead."));
```
Please also add '[deprecated]' at the end of the flag description string.

wenzeslaus commented:
Do we need the connection info in the output? Or in other words, what is the reason for including it?

gulshan-123 commented:
I have added the JSON support for -d flag

It prints some new fields not present in `info` section of output without `-d` flag:
ncols, nrows, length, position.

```json
{
    "table": "firestations",
    "ncols": 5,
    "nrows": 4,
    "columns": [
        {
            "length": 20,
            "position": 1,
            "name": "cat",
            "sql_type": "INTEGER",
            "is_number": true
        },
        {
            "length": 1000,
            "position": 2,
            "name": "agency",
            "sql_type": "TEXT",
            "is_number": false
        },
        {
            "length": 1000,
            "position": 3,
            "name": "notes",
            "sql_type": "TEXT",
            "is_number": false
        },
        {
            "length": 20,
            "position": 4,
            "name": "score",
            "sql_type": "REAL",
            "is_number": true
        },
        {
            "length": 1000,
            "position": 5,
            "name": "whenCreated",
            "sql_type": "TEXT",
            "is_number": false
        }
    ]
}
```

petrasovaa commented on db/db.select/main.c:
```diff
@@ -302,6 +499,23 @@ void parse_command_line(int argc, char **argv)
+        G_verbose_message(
+            _("Flag 'v' is deprecated and will be removed in a future "
+              "release. Please use format=vertical instead [deprecated]."));
+        if (format->answer && parms.format != VERTICAL) {
```
What I meant to was putting the [deprecated] at the end of the description on line 420.
